ui/org.eclipse.pde.core/src/org/eclipse/pde/core/project/IPackageExportDescription.java

```diff
@@ -29,28 +31,54 @@ public interface IPackageExportDescription {
 	 * Returns the fully qualified name of the exported package.
 	 *
 	 * @return fully qualified name of the exported package
+	 * @since 3.19
 	 */
-	public String getName();
+	String name();
+
+	/** @deprecated Instead use {@link #name()} */
+	@Deprecated(since = "3.19")
+	default String getName() {
+		return name();
+	}
 
 	/**
-	 * Returns the version of the exported package or <code>null</code>
-	 * if unspecified.
+	 * Returns the version of the exported package or <code>null</code> if
+	 * unspecified.
 	 *
 	 * @return version or <code>null</code>
+	 * @since 3.19
+	 */
+	Version version();
+
+	/** @deprecated Instead use {@link #version()} */
+	@Deprecated(since = "4.19")
+	default Version getVersion() {
+		return version();
+	}
+
+	/**
+	 * Returns the declared friends of this package.
+	 *
+	 * @return friends as bundle symbolic names, may be empty
+	 * @since 3.19
 	 */
-	public Version getVersion();
+	List<String> friends();
```

Comment:
```suggestion
	Stream<String> friends();
```

I usually prefer to return streams instead of collections, this has many benefits:

1. One has not to define if collection is modifiable or not
2. No need to think about the actual collection type consumers can collect to any type they want
3. easier to use in streaming / filtering if the caller further wants to customize

Reply:
The downside of `Stream` for API signature is its stateful nature, callers should be ready to it.

Reply:
Of course each call must return a _new stream_  that is not shared with others...

Reply:
Yes, sure.
I rather mean that callers should be aware that they need to use it much more carefully than collection: do not try to call terminal operation more than once and so on. And this constraint may require to change programming style a bit. 
Personally I'm fine with it, but the rest of PDE code base uses an ancient style full of static procedures, local variables and highly mutable data structures.

Reply:
We try to getting better every day :-)

Reply:
I'm personally not a fan of mixing all kinds of styles.   That's just my opinion...

Reply:
While Streams are often useful and suitable, e.g. in https://github.com/eclipse-pde/eclipse.pde/pull/1163, I'm not convinced they suite here because this interface models a package export. From that POV the value of the `x-friends` directive is a List. And these model objects are also created from a List, so I think it is fine to model it as a list again.

>     1. One has not to define if collection is modifiable or not
>
With modern Java one should not assume to get mutable collections unless it is explicitly stated by the doc. In this consumers will notice it latest on their first attempt that the returned list is immutable. But if you think it's necessary to document it, I can add it.

>     2. No need to think about the actual collection type consumers can collect to any type they want
>
As described above, in this case I think this is implied by what is modeled. 

>     3. easier to use in streaming / filtering if the caller further wants to customize

A stream is just one `.stream()` away and the creation of other collection types is also easy since most collections provide a constructor that accepts arbitrary collection types.

Reply:
> From that POV the value of the x-friends directive is a List

Really? Does order matter? Do we need to access them by index? Are duplicate entries allowed? Why then not keep `String[]`? If not at least the return type should be Collection<String>

> A stream is just one .stream() away and the creation of other collection types is also easy since most collections provide a constructor that accepts arbitrary collection types.

The point is that usually inside such methods one uses stream and then just collect them into a List  and the caller is then converting the collected list into a stream just to do something with it and collect it into some other kind of collection (or a new list).

Also Streams are lazy evaluate, so if I'm only interested in the first element I only need to possibly construct one element while when returning a collection I always need to collect / construct the whole collection.

This does not apply here but should be kept in mind when modernizing APIs, a stream is often the better choice on the long run.

Reply:
> > From that POV the value of the x-friends directive is a List
> 
> Really? Does order matter? Do we need to access them by index? Are duplicate entries allowed? Why then not keep `String[]`? If not at least the return type should be Collection

OK, I was just thinking about the directive but in the PDE model duplicates are indeed not permitted. In the UI the friends are even always sorted. So it is actually a Set and could even be a `SortedSet`, although the friends values are not persisted in sorted order in the manifest.

Arrays break the immutability and don't play well with recoreds since the default equals for arrays is an identity check.

> 
> > A stream is just one .stream() away and the creation of other collection types is also easy since most collections provide a constructor that accepts arbitrary collection types.
> 
> The point is that usually inside such methods one uses stream and then just collect them into a List and the caller is then converting the collected list into a stream just to do something with it and collect it into some other kind of collection (or a new list).
> 
> Also Streams are lazy evaluate, so if I'm only interested in the first element I only need to possibly construct one element while when returning a collection I always need to collect / construct the whole collection.
> 
> This does not apply here but should be kept in mind when modernizing APIs, a stream is often the better choice on the long run.

I think that the Stream API is a great 'addition/extension' of the Collections API, but I don't expect that it will entirely replace collections. The point for me is, that we have a backing collection for the `friends` (as we can obviously not use a stream). Therefore I don't see a problem to choose its type wisely and expose the suitable type in the API. If that's sufficient for a caller: great. And if not the caller still can transform it. Always creating a stream is just a waste of resources for those that are fine with the chosen type, as we don't perform any transformation here and perform some lazy operation.